src/deploy/dataconnect/prepare.ts

```diff
@@ -55,6 +58,37 @@ export default async function (context: any, options: Options): Promise<void> {
     filters,
   };
   utils.logLabeledBullet("dataconnect", `Successfully prepared schema and connectors`);
+  if (options.dryRun) {
+    utils.logLabeledBullet("dataconnect", "Checking for CloudSQL resources...");
+    return;
+  }
+  await Promise.all(
+    serviceInfos
+      .filter((si) => {
+        return !filters || filters?.some((f) => si.dataConnectYaml.serviceId === f.serviceId);
+      })
+      .map(async (s) => {
+        const instanceId = s.schema.primaryDatasource.postgresql?.cloudSql.instance
+          .split("/")
+          .pop();
+        const databaseId = s.schema.primaryDatasource.postgresql?.database;
+        if (!instanceId || !databaseId) {
+          return Promise.resolve();
+        }
+        const enableGoogleMlIntegration = requiresVector(s.deploymentMetadata);
+        utils.logLabeledBullet("dataconnect", "Checking for CloudSQL resources...");
+
+        return provisionCloudSql({
+          projectId,
+          locationId: parseServiceName(s.serviceName).location,
+          instanceId,
+          databaseId,
+          enableGoogleMlIntegration,
+          waitForCreation: true,
+          dryRun: options.dryRun,
+        });
+      }),
+  );
```

Comment:
This `await` section looks duplicated from deploy.ts, should it be removed there?